### PR TITLE
Config chapter rework

### PR DESCRIPTION
**Pull request by asmelko**

### CODE REVIEW

File: user_documentation/configuration.tex [hunk 1]
```diff
@@ -1,11 +1,22 @@
 \section{External Macro Libraries and COPY Members}
 \label{sec:configuration}
 
-The HLASM Language Support extension looks for locally stored members when a macro or COPY instruction is evaluated. The paths of these members are specified in two configuration files in the \TT{.hlasmplugin} folder of the currently open workspace. Ensure that you configure these files before using macros from separate files or the COPY instruction.
+The HLASM Language Support extension looks for locally stored members when a macro or COPY instruction is evaluated. The paths of these members are specified in two configuration files in the \TT{.hlasmplugin} folder of the currently open workspace:
+\begin{itemize}
+	\item  \TT{proc\_grps.json} defines \textbf{processor groups} by assigning a group name to a list of directories. Hence, the group name serves as an unique identifier of a HLASM library set defined by a list of directories.
```
Comment (marcel1hruska) [inline user_documentation/configuration.tex]:
```suggestion
	\item  \TT{proc\_grps.json} defines \textbf{processor groups} by assigning a group name to a list of directories. Hence, the group name serves as a unique identifier of a HLASM library set defined by a list of directories.
```

Comment (marcel1hruska) [reply user_documentation/configuration.tex]: HLASM library set? Ako mnozina HLASM kniznic? Ak hej, otocil by som to - 'a set of HLASM libraries'.

File: user_documentation/configuration.tex [hunk 2]
```diff
@@ -1,11 +1,22 @@
...
+	
+	
+	\item  \TT{pgm\_conf.json} provides a mapping between \textbf{programs} (open-code files) and processor groups. It specifies which list of directories is used with which source file. If a relative source file path is specified, it is relative to the current workspace.
+\end{itemize}
+
+Therefore, when you want to use a predefined set of macro and copy members in your source code, you do:
```
Comment (marcel1hruska) [inline user_documentation/configuration.tex]:
```suggestion
Therefore, to use a predefined set of macro and copy members, do the following steps: 
```

File: user_documentation/configuration.tex [hunk 3]
```diff
@@ -56,22 +62,48 @@ \section{External Macro Libraries and COPY Members}
...
+An example configuration can be seen in \cref{lst:ex1} and \cref{lst:grps}. It describes two processor groups \TT{GROUP1}, \TT{GROUP2} and their respective mappings to programs \TT{source} and \TT{asm\_prg}. If you have the two configuration files configured as above and invoke the MAC1 macro from \TT{source}, the folder \TT{ASMMAC/} in the current workspace is searched for a file named exactly MAC1. If that file is not found, the folder \TT{C:/SYS.ASMMAC} is searched. If that search is unsuccessful too, an error displays that the macro does not exist.
+
+Note that the macro \TT{MAC1} is searched in directories in order as they are listed in the configuration. 
+
+To add a possibility to specify a custom extension for HLASM files in the workspace, {pgm\_conf.json} has an optional field \TT{alwaysRecognize}. It takes an array of wildcards and provides the following functionality:
 \begin{itemize}
-	\item All files matching these wildcards will always be recognized as HLASM files. 
+	\item All files matching these wildcards will always be recognized as HLASM files. Hence, the HLASM plugin will be automatically activated on all the matched files.
```
Comment (marcel1hruska) [inline user_documentation/configuration.tex]: Not true really... On bezi stale :D

File: user_documentation/configuration.tex [hunk 4]
```diff
@@ -56,22 +62,48 @@ \section{External Macro Libraries and COPY Members}
...
 	\item If an extension wildcard is defined, all macro and copy files with such extension may be used in the source code. For example, with the extension wildcard \TT{*.hlasm}, a user may add macro MAC to his source code even if it is in a file called \TT{Mac.hlasm}.
 \end{itemize}
 
-\begin{listing}[H]
+\Cref{lst:alw} demonstrates the mentioned functionality. With this configuration file, processor group \TT{GROUP1} will be assigned to \TT{source} and \TT{source.hlasm} file as well. Also, macro and copy files in the \TT{lib} directory can be referenced in the program without the \TT{.asm} extension  but still be correctly recognized.
```
Comment (marcel1hruska) [inline user_documentation/configuration.tex]:
```suggestion
\Cref{lst:alw} demonstrates the mentioned functionality. With this configuration file, processor group \TT{GROUP1} will be assigned to \TT{source} and \TT{source.hlasm} file as well. Also, macro and copy files in the \TT{lib} directory can be referenced and correctly recognized in the program without the \TT{.asm} extension.
```

File: user_documentation/start.tex [hunk 5]
```diff
@@ -26,15 +26,17 @@ \subsection{Usage}
...
-	\item If the auto-detection of HLASM language fails with the file, set it manually in the bottom-right corner of the VS Code window.  
+	\item If the auto-detection of HLASM language does not recognize the file, set it manually in the bottom-right corner of the VS Code window.  
 	\item The extension is now enabled on the opened file. If you have macro definitions in separate files or use the COPY instruction, you need to setup the workspace.
 \end{enumerate}
 
 \subsection{Setting up a multi-file project environment}
-HLASM COPY instruction copies the source code from various external files, as driven by HLASM evaluation. The source code interpreter in the HLASM Extension needs to be set up correctly to be able to find the same files as the HLASM assembler program. This is done by setting up two files that describe HLASM program config and processor groups:
+HLASM COPY instruction copies the source code from various external files, as driven by HLASM evaluation. The source code interpreter in the HLASM Extension needs to be set up correctly to be able to find the same files as the HLASM assembler program.
```
Comment (marcel1hruska) [inline user_documentation/start.tex:34]: Tu nemusi byt jasne ako sa nastavit source code interpreter.

File: user_documentation/configuration.tex (hunk 1 above)
Comment (asmelko) [reply user_documentation/configuration.tex]: jop

File: user_documentation/configuration.tex (hunk 3 above)
Comment (asmelko) [reply user_documentation/configuration.tex]: chcel som tym viac zdoraznit, ze budes na tom file moct pouzivat funkcionalitu nasho celeho pluginu. ako by som to mal preformulovat?

File: user_documentation/start.tex [hunk 6]
```diff
@@ -26,15 +26,17 @@ \subsection{Usage}
...
+
+This is done by setting up two configuration files --- \TT{proc\_grps.json} and \TT{pgm\_conf.json}. The extension guides the user in their creation:
 \begin{enumerate}
-	\item After opening a HLASM file for the first time, two popups display. Select \TT{Create pgm\_conf.json with current program} and \TT{Create empty proc\_grps.json}. The two configuration files will be created in the \TT{.hlasmplugin} subfolder.
-	\item In the \TT{proc\_grps.json} file, fill the \TT{libs} array with paths to folders with macro definitions and COPY files. For example, if you have your macro files in the \TT{ASMMAC/} folder, add the string \TT{"ASMMAC"} into the libs array.
+	\item After opening a HLASM file for the first time, two popups display. Select \TT{Create pgm\_conf.json with current program} and \TT{Create empty proc\_grps.json}. The two configuration files are then created with default values. They are written into the \TT{.hlasmplugin} subfolder.
```
Comment (marcel1hruska) [inline user_documentation/start.tex]:
```suggestion
	\item After opening a HLASM file for the first time, two pop-ups are displayed. Select \TT{Create pgm\_conf.json with current program} and \TT{Create empty proc\_grps.json}. The two configuration files are then created inside the \TT{.hlasmplugin} subfolder with default values.
```

File: user_documentation/configuration.tex (hunk 3 above)
Comment (marcel1hruska) [reply user_documentation/configuration.tex]: Jasne, chapem. Mozno nahradit activated za running? Alebo "the HLASM plugin will be automatically processing all matched files"?

**Review by marcel1hruska — APPROVED** [review]

**Review by michalbali256 — APPROVED** [review]

**Pull request merged**